Git Export Wizard Validation And Next Button Enablement

elvisisking opened the pull request:
- Added field validation for git credentials control. Error messages are displayed underneath the field that has an error.
- Git export wizard next button is disabled when field validation fails.
- Layout changes to repository information section of the git credentials control.

Issue: The git credentials control repository properties border title css markup needs work so that the border does not appear behind the title.
Issue: Since ng-message needs an ng-model to work, and input form controls that have a type of file do not have an ng-model, the validation logic does not work when those input controls are involved.
Issue: Seems like the export wizard height should be a bit larger. I think there is a max-height somewhere that controls this.

elvisisking committed:
- Added field validation for git credentials control. Error messages are displayed underneath the field that has an error.
- Git export wizard next button is disabled when field validation fails.
- Layout changes to repository information section of the git credentials control.

Issue: The git credentials control repository properties border title css markup needs work so that the border does not appear behind the title.
Issue: Since ng-message needs an ng-model to work, and input form controls that have a type of file do not have an ng-model, the validation logic does not work when those input controls are involved.
Issue: Seems like the export wizard height should be a bit larger. I think there is a max-height somewhere that controls this.

phantomjinx commented on vdb-bench-assembly/plugins/vdb-bench-dataservice/export/dsExportGitWizard.js:
```diff
@@ -90,7 +103,25 @@
+            
+            // see if SSH authentication is complete
+            if ( !_.isEmpty( vm.repo.parameters[ 'repo-password-property' ] ) && !_.isEmpty( vm.repo.parameters[ 'repo-known-hosts-property' ] ) ) {
+              return true;
```
These validation tests do not make sense. A git repository can use http authentication, ssh authentication and even have no authentication at all. As such, I think it difficult to try to assume which of these properties should be populated. For example, it is 'normal' for the known-hosts to be required for ssh but if the ssh server is configured to not use known-hosts then this is unnecessary. Thus, my recommendation would be to remove these tests at this time.

elvisisking replied:
I'll remove the authentication checks.

phantomjinx commented on vdb-bench-assembly/plugins/vdb-bench-dataservice/export/dsExportGitWizard.js:
```diff
@@ -90,7 +103,25 @@
 
-            return true;
+            if ( $rootScope.requireAuthorName && _.isEmpty(vm.repo.parameters['author-name-property'])) {
+                return false;
```
I see the sense of doing this and understand how it solves the problem. Yet I can still hear my former programming tutor at uni declaring me as Satan for suggesting the use of global variables. Maybe we can stick a TODO comment around these to maybe revisit in the future to see if there is an alternative way?

elvisisking replied:
Instead of using a global variable I will use an event via $scope.$emit.

phantomjinx commented on vdb-bench-assembly/plugins/vdb-bench-widgets/gitCredentialsControl.js:
```diff
@@ -36,8 +43,19 @@
         var vm = this;
+        vm.protocol = {
+            "useHttp": true,
+            "useSsh": false
```
What about using neither? This is not a daft question since a git repository with no authentication could be added onto the local filesystem, eg. I run my server locally and also create a git repo on my own machine to get the artifacts out of the workspace and into a version-controlled repo. This is a major possibility and a viable use-case.

elvisisking replied:
I'll remove this variable and all references to it.

phantomjinx commented on vdb-bench-assembly/plugins/vdb-bench-widgets/gitCredentialsControl.js:
```diff
@@ -70,6 +88,34 @@
+                return false;
+            }
+
+            return !_.isEmpty( vm.repo.parameters[ 'repo-password-property' ] ) && !_.isEmpty( vm.repo.parameters[ 'repo-known-hosts-property' ] );
```
Again questionable whether known-hosts is a required pre-requisite for ssh. Its required on my ssh configuration but no guarantee its essential for all setups.

elvisisking replied:
I'll remove the authentication checks and checks for known hosts file.

elvisisking committed:
Made changes based on recommendations.
- No longer using global variables. Instead using events via $scope.$emit.
- Removed validation of the SSH and HTTP input controls.
- Removed validation of repository authentication parameters.
- Removed i18n messages that are no longer needed.

elvisisking commented:
My last commit @phantomjinx reflects the changes I made based on your recommendations.
